src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/TypeInfos/RuntimeTypeInfo.GetMember.cs

```diff
@@ -123,6 +123,21 @@ public sealed override MemberInfo GetMemberWithSameMetadataDefinitionAs(MemberIn
             if (member is null)
                 throw new ArgumentNullException(nameof(member));
 
+            // Need to walk up the inheritance chain if member is not found
+            // Leverage the existing cache mechanism of per type to store members
+            RuntimeTypeInfo? runtimeType = this;
+            while (runtimeType != null)
+            {
+                MemberInfo result = runtimeType.GetMemberWithSameMetadataDefinitionAsInternal(member);
+                if (result != null)
+                    return result;
+                runtimeType = runtimeType.BaseType as RuntimeTypeInfo;
+            }
+            throw new ArgumentException(SR.Format(SR.Arg_MemberInfoNotFound, member.Name), nameof(member));
+        }
+
+        private MemberInfo GetMemberWithSameMetadataDefinitionAsInternal(MemberInfo member)
```

Comment:
I do not think you need to extract this into a separate method. It can be inline in the while loop.